### DISCUSSION TITLE
迁移到其他博客系统?

### DISCUSSION THREAD

**Discussion by zkz098** [theme-shoka-x/hexo-theme-shokaX#162] [Ideas]
这是一个长期计划，不保证实现。
旨在于将 ShokaX 迁移到astro等现代博客系统，优化开发效率和性能，解决长期技术债务。这并不意味着 Hexo 版本停更，目前的设想如下:
ShokaX Assets: 跨平台CSS和Typescript文件
ShokaX hexo/astro/... theme: 基于Assets构建的各系统独立版本
同时对ShokaX进行Vue化，如果有需求Vitepress/Vuepress也不是不行（
主要原因是现有版本技术债务有点离谱了，Markdown渲染器翻新很难出成效但费时费力，现有Typescript又难以和现代框架结合，只能大规模重写底层

另外Home界面是否需要改动以更适应现代审美，感觉 ShokaX 的主界面略显单调

目前计划为未来1-2年迁移到astro，可能更慢

> 昨天人有点晕，让AI代笔了一下，结果问题很大，今天重写了一份

**Comment by zkz098** [comment]
昨天人有点晕，让AI代笔了一下，结果问题很大，今天重写了一份

**Comment by D-Sketon** [comment]
有无向hugo或halo迁移的打算（）

**Reply by zkz098** [reply]
有，至少halo有（）

**Comment by zkz098** [comment]
目前看来 astro 版使用 react 重写 ts 部分可能是最佳选择。
可能会使用 Tailwind 等 CSS 框架重写样式表

**Reply by zkz098** [reply]
开个投票确定一下技术选型，这个影响有点大（
